backend/src/permissions.ts

```diff
@@ -27,6 +27,10 @@ export class PermissionsManager {
     return mem.role === 'lead' || this.isAdmin(mem);
   }
 
+  static async canEditTeamEvent(mem: IdolMember): Promise<boolean> {
+    return mem.role === 'lead' || this.isAdmin(mem);
+  }
+
```

Comment:
we should probably also consider refactoring the permissions manager since `mem.role === 'lead' || this.isAdmin(mem)` is just copied a bunch of times but probably don't have to worry about that in this PR